R/opts.R

```diff
@@ -678,33 +765,57 @@ rstan_opts <- function(object = NULL,
 #' @param return_fit Logical, defaults to TRUE. Should the fit stan model be
 #' returned.
 #'
-#' @param ... Additional parameters to pass  underlying option functions.
+#' @param ... Additional parameters to pass to underlying option functions,
+#'   [stan_sampling_opts()] or [stan_vb_opts()], depending on the method
 #'
 #' @importFrom rlang arg_match
-#' @return A `<stan_opts>` object  of arguments to pass to the appropriate
+#' @return A `<stan_opts>` object of arguments to pass to the appropriate
 #' rstan functions.
 #' @author Sam Abbott
+#' @author Sebastian Funk
 #' @export
 #' @inheritParams rstan_opts
-#' @seealso [rstan_opts()]
+#' @seealso [stan_sampling_opts()] [stan_vb_opts()]
 #' @examples
 #' # using default of [rstan::sampling()]
 #' stan_opts(samples = 1000)
 #'
 #' # using vb
 #' stan_opts(method = "vb")
-stan_opts <- function(samples = 2000,
+stan_opts <- function(object = NULL,
+                      samples = 2000,
+                      method = "sampling",
                       backend = "rstan",
                       init_fit = NULL,
                       return_fit = TRUE,
                       ...) {
-  backend <- arg_match(backend, values = "rstan")
-  if (backend == "rstan") {
-    opts <- rstan_opts(
-      samples = samples,
-      ...
+  method <- arg_match(method, values = c("sampling", "vb"))
```

Comment:
Given these are checked here my other suggestions may not really be needed but it also seems like it can't hurt.